homeassistant/components/tuya/light.py

```diff
@@ -542,7 +542,7 @@ def turn_on(self, **kwargs: Any) -> None:
 
         if (
             ATTR_BRIGHTNESS in kwargs
-            and self.color_mode != ColorMode.HS
+            and ATTR_HS_COLOR not in kwargs
```

Comment:
This change is not correct. In case a service call is made that just changes brightness (thus no `ATTR_HS_COLOR` in the kwargs), while still the hs color mode isn't handled by this call (as brightness is part of the color).

../Frenck

Reply:
I will perform some further testing and note what I tested but I'm pretty sure that I tested the scenario where only the brightness was changed and there was no issue caused concerning erroneous changing of the work mode. 

I can perform some further testing and note the combinations I perform, and report back here. However, if there's a better way to show my testing, I'd be happy to go through the steps required to perform the more in-depth methods of testing.

Reply:
Okay so it turns out I did not test the actual use case that I saw the original code checking for in the brightness section: Changing only the brightness while in colour work mode. 

Instantly set the work mode back to white like ya'll predicted. My apologies for the mistake.  I like @emontnemery's solution so I will give that a shot

Reply:
[9177c61](https://github.com/home-assistant/core/pull/88470/commits/9177c61147cfebf4bccd67b1b2ee9bac8fb61b3e) resolves the issue + made sure to note my tests. The checkmark indicates that the tests behaved as expected.